hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -142,9 +158,14 @@ public void triggerContainerHealthCheck() {
               " processing {} containers.", Time.monotonicNow() - start,
           containers.size());
       logUnhealthyContainerStats(unhealthyContainerStateStatsMap);
-      processedContainers.clear();
-    } finally {
-      lock.writeLock().unlock();
+      if (containers.size() > Integer.parseInt(DEFAULT_FETCH_COUNT)) {
+        startID = ContainerID.valueOf(
+            containers.get(containers.size() - 1).getContainerID());
+        containers.clear();
+        containers = containerManager.getContainers(startID,
+            Integer.parseInt(DEFAULT_FETCH_COUNT));
+      }
+      containers.clear();
```

Comment:
The `containers.clear()`; statement inside the loop is meant to clear the list before fetching the next batch. However, the last `containers.clear();` statement outside the loop seems redundant since the loop condition `(while (!containers.isEmpty()))` ensures that it's only executed when containers is empty.?

Reply:
@ArafatKhan2198 I couldn't understand this comment, as I don't see `containers.clear()`  outside the loop. Can you pls clarify ?

Reply:
Apologies for the earlier mistake in the comment. I have now corrected it. Please take a look.

Reply:
As discussed, we have a case where last iteration in batch will not execute the if condition , so need to clear -off any in memory.